javascript/test/test_refactor.js

```diff
@@ -1,621 +1,971 @@
-         engine.flagAPIToLiteral();
-         engine.finalizeLiterals();
-
-         assert(ast.body[0].expression.createdByPiranha === undefined);
-      });
-   })
-})
-
+describe("refactor", () => {
+  describe("#flagAPIToLiteral", () => {
+    it("should introduce the createdByPiranha property on the literals", () => {
+      const code = "isFlagTreated(testFlag)";
+
+      let ast = recast.parse(code).program;
+      const engine = new refactor.RefactorEngine(
+        ast,
+        properties,
+        treated,
+        flagname
+      );
+      engine.flagAPIToLiteral();
+
+      assert(ast.body[0].expression.createdByPiranha !== undefined);
+    });
+
+    it("should replace calls to flag check APIs with boolean literals", () => {
+      const code = `if (isFlagTreated(testFlag)) {} else {}`;
+      const expected_code = `if (true) {} else {}`;
+      const ast = recast.parse(code).program;
+
+      const engine = new refactor.RefactorEngine(
+        ast,
+        properties,
+        treated,
+        flagname
+      );
+      engine.flagAPIToLiteral();
+      engine.finalizeLiterals();
+
+      const refactored_code = recast.print(ast).code;
+
+      assert(engine.changed);
+      assert(
+        expected_code === refactored_code,
+        `\nEXPECTED : ${JSON.stringify(
+          expected_code
+        )}\nREFACTORED : ${JSON.stringify(refactored_code)}`
+      );
+    });
+  });
+
+  describe("#evalBoolExpressions", () => {
+    it("should replace true && f() with just f()", () => {
+      const code = `if (isFlagTreated(testFlag) && f()) {} else {}`;
+      const expected_code = `if (f()) {} else {}`;
+      const ast = recast.parse(code).program;
+
+      const engine = new refactor.RefactorEngine(
+        ast,
+        properties,
+        treated,
+        flagname
+      );
+      engine.flagAPIToLiteral();
+      engine.evalBoolExpressions();
+      engine.finalizeLiterals();
+
+      const refactored_code = recast.print(ast).code;
+
+      assert(engine.changed);
+      assert(
+        expected_code === refactored_code,
+        `\nEXPECTED : ${JSON.stringify(
+          expected_code
+        )}\nREFACTORED : ${JSON.stringify(refactored_code)}`
+      );
+    });
+
+    it("should replace false && f() with just false", () => {
+      const code = `if (isToggleDisabled(testFlag) && f()) {} else {}`;
+      const expected_code = `if (false) {} else {}`;
+      const ast = recast.parse(code).program;
+
+      const engine = new refactor.RefactorEngine(
+        ast,
+        properties,
+        treated,
+        flagname
+      );
+      engine.flagAPIToLiteral();
+      engine.evalBoolExpressions();
+      engine.finalizeLiterals();
+
+      const refactored_code = recast.print(ast).code;
+
+      assert(engine.changed);
+      assert(
+        expected_code === refactored_code,
+        `\nEXPECTED : ${JSON.stringify(
+          expected_code
+        )}\nREFACTORED : ${JSON.stringify(refactored_code)}`
```

Comment:
Prettier may refactor some code in undesirable ways. For example, this way of writing the output string seems to be unnecessarily confusing.